v1/format/format.go

```diff
@@ -1207,10 +1207,19 @@ func (w *writer) writeTermParens(parens bool, term *ast.Term, comments []*ast.Co
 	case ast.String:
 		if term.Location.Text[0] == '`' {
 			// To preserve raw strings, we need to output the original text,
-			// not what x.String() would give us.
 			w.write(string(term.Location.Text))
 		} else {
-			w.write(x.String())
+			// x.String() cannot be used by default because it can change the input string "\u0000" to "\x00"
+			// term.Location.Text could contain more than just the string, extract the first string it can find
+			_, after, found := strings.Cut(string(term.Location.Text), "\"")
+			if !found {
+				// If no quoted string was found, that means it is a key being formatted to a string
+				// e.g. partial_set.y to partial_set["y"]
+				w.write(x.String())
+			} else {
+				w.write("\"" + after)
```

Comment:
What if we have a raw string containing `"`? 
E.g. 

```
else := `{"foo":"bar"}`
```

Reply:
Handling of raw strings is unchanged, there is an existing check right before to detect raw strings that just uses the raw text:

```
if term.Location.Text[0] == '`' {
	// To preserve raw strings, we need to output the original text,
	w.write(string(term.Location.Text))
} else {
```

So should work the same as before

Reply:
That's only true if the raw string backtick is the first token; not if it's part of an `else`-head:

```rego
a := "foo" if {
	false
} else := `{"foo":"bar"}`
```